packages/a11y-base/src/aria-hidden.js

```diff
@@ -60,13 +54,12 @@ const correctTargets = (parent, targets) => {
         return null;
       }
 
-      if (parent.contains(target)) {
-        return target;
-      }
-
-      const correctedTarget = unwrapHost(target);
-      if (correctedTarget && parent.contains(correctedTarget)) {
-        return correctedTarget;
+      let node = target;
+      while (node && node !== parent) {
+        if (parent.contains(node)) {
+          return target;
+        }
+        node = node.parentNode || node.host;
```

Comment:
I wonder if the solution could be optimized by always skipping to the root element instead of traversing all the parent elements.

```suggestion
node = node.getRootNode().host;
```

Reply:
Good point, thanks. Yes, it can be done this way.